packages/gatsby-transformer-screenshot/src/gatsby-node.js

```diff
@@ -68,17 +68,20 @@ exports.onPreBootstrap = (
   })
 }
 
-exports.onCreateNode = async ({
-  node,
-  actions,
-  store,
-  cache,
-  createNodeId,
-}) => {
+exports.onCreateNode = async (
+  { node, actions, store, cache, createNodeId },
+  pluginOptions
+) => {
   const { createNode, createParentChildLink } = actions
 
-  // We only care about parsed sites.yaml files with a url field
-  if (node.internal.type !== `SitesYaml` || !node.url) {
+  /*
+   * Check if node is of a type we care about, and has a url field
+   * (originally only checked sites.yml, hence including by default)
+   */
+  if (
+    ![`SitesYaml`, ...pluginOptions.nodeTypes].includes(node.internal.type) ||
```

Comment:
Also I'd maybe say it's worth re-writing like this for clarity?

```js
const validNodeTypes = [`SitesYaml`].concat(pluginOptions.nodeTypes || []);

if (!validNodeTypes.includes(node.internal.type) || !node.url) {
  return
}
```

Reply:
I like it